test/test_nn.py

```diff
@@ -5783,6 +5783,39 @@ def test_Transformer_cell(self):
                              memory_key_padding_mask=memory_key_padding_mask)
         output.sum().backward()
 
+    def test_Transformer_cell_batch_first(self):
```

Comment:
Cool. Now `for batch_first in (True, False):` blocks are used.

Reply:
Now the three errors in the `test_nn.py` are fixed, as I have trouble running directly the `test_nn.py` on my end due to VScode complained submodules not being loaded. I created a minimal environ to make sure the errors are fixed: https://github.com/scaomath/torch-playground

Reply:
I felt that this error is on the verge to be a non-rounding error. On my end, using the original `test_nn.py`, then test cannot pass rel tol 1e-7 and abs tol 1e-5, this is the maximum diff for each entries of `result` and `ref_output` in each test of `test_transformerdecoder`:
```
0.00012551248
1.7166138e-05
8.940697e-07
9.536743e-07
9.536743e-07
9.536743e-07
4.7683716e-07
0.00015902519
2.861023e-06
9.2715025e-05
4.053116e-06
0.00016299635
2.2487715e-05
4.7683716e-07
3.091991e-07
```

Since `test_transformerdecoderlayer` has no problem on `batch_first in (False, True)` loop, I feel the `test_transformerdecoder` cannot pass the test due to other changes.

Reply:
Okay, culprit found when I am preparing my class this morning. It is not reasonable to set the same absolute tolerance for inputs with different magnitudes. For example, the input matrix has entries all between 0 and 1, vs another input that ranges from 40 to 80, of course we are gonna get `O(10^2)` rounding error margin. 

Now in the `test_transformerdecoder`, `rtol` is unchanged `1e-7`, but `atol` scales with the input, if the inputs are from 10-20, then `atol` is dialed down from `1e-5` to `1e-4`, if the inputs are in the range of 50-80, the `atol` is further reduced to `1e-3`. The relative tolerance tells the story.

@jbschlosser @zhangguanheng66 @albanD